public/video-ui/src/components/VideoPreview/VideoPreview.js

```diff
@@ -1,33 +1,25 @@
 import React from 'react';
+import { VideoEmbed } from '../utils/VideoEmbed';
 import { YouTubeEmbed } from '../utils/YouTubeEmbed';
 
 export default class VideoPreview extends React.Component {
-  getActiveAsset = () => {
-    if (!this.props.video.assets || !this.props.video.activeVersion) {
-      return false;
-    }
-
-    for (let i = 0; i < this.props.video.assets.length; i++) {
-      if (
-        this.props.video.activeVersion === this.props.video.assets[i].version
-      ) {
-        return this.props.video.assets[i];
-      }
-    }
-  };
-
   renderPreview() {
-    const activeAsset = this.getActiveAsset();
+    const { assets, activeVersion } = this.props.video;
+    const relevant = assets.filter(asset => asset.version === activeVersion);
 
-    if (!activeAsset) {
+    if (!assets || !activeVersion || relevant.length === 0) {
```

Comment:
If the assets do not exist (as is suggested here that they may not), will we get an exception from line 8?